plugin/route53/setup.go

```diff
@@ -26,8 +26,8 @@ var log = clog.NewWithPlugin("route53")
 func init() { plugin.Register("route53", setup) }
 
 // exposed for testing
-var f = func(credential *credentials.Credentials) route53iface.Route53API {
-	return route53.New(session.Must(session.NewSession(&aws.Config{Credentials: credential})))
+var f = func(credential *credentials.Credentials, endpoint *string) route53iface.Route53API {
```

Comment:
nit: doesn't seem like it _needs_ to be a pointer since you're always passing empty string as default?

Reply:
hm actually looks like both nil and empty string are acceptible as defaults per aws docs.